enterprise/ha/src/main/java/org/neo4j/kernel/ha/lock/SlaveLocksClient.java

```diff
@@ -185,52 +198,58 @@ public void releaseExclusive( Locks.ResourceType resourceType, long resourceId )
     }
 
     @Override
-    public void releaseAll()
+    public void stop()
+    {
+        if ( txTerminationAwareLocks )
+        {
+            client.stop();
+            endLockSessionOnMaster( false );
+            stopped = true;
+        }
+        else
+        {
+            throw new UnsupportedOperationException( "Lock client stop is unsupported on slave side." );
```

Comment:
Wouldn't this be better as a no-op? I don't see how other lock clients throw exception if tx termination awareness is disabled.